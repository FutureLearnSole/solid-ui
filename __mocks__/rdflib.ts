export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    doc: () => {
      equals: () => {}
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {

}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
}
