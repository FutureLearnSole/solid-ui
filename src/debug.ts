export const debug = {
  log (...args: any[]) {
    // eslint-disable-next-line no-console
    console.log(...args)
  },
  warn (...args: any[]) {
    // eslint-disable-next-line no-console
    console.warn(...args)
  },
  error (...args: any[]) {
    // eslint-disable-next-line no-console
    console.error(...args)
  },
  trace (...args: any[]) {
    // eslint-disable-next-line no-console
    console.trace(...args)
  }
}
