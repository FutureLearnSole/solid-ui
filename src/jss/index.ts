import { create, Jss, Styles, StyleSheet } from 'jss'
import preset from 'jss-preset-default'

const stylesheetsMap = new Map<HTMLElement, Jss>()

export function getClasses (insertionPoint: HTMLElement, styles: Partial<Styles>): StyleSheet {
  const stylesheet = getStylesheet(insertionPoint)
  return stylesheet.createStyleSheet(styles).attach()
}

export function getStylesheet (insertionPoint: HTMLElement): Jss {
  const cachedStylesheet = stylesheetsMap.get(insertionPoint)
  if (cachedStylesheet) {
    return cachedStylesheet
  }
  const stylesheet = create({
    insertionPoint,
    plugins: preset().plugins
  })
  stylesheetsMap.set(insertionPoint, stylesheet)
  return stylesheet
}
