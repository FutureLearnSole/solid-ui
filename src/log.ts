import { escapeForXML } from './utils'

// ///////////////////////  Logging
//
// bitmask levels
// const TNONE = 0
const TERROR = 1
const TWARN = 2
const TMESG = 4
const TSUCCESS = 8
const TINFO = 16
const TDEBUG = 32
const TALL = 63

/* global alert */
class Log {
  public level: number = TERROR + TWARN + TMESG
  public ascending: boolean = false

  public msg (str: string, type: number = TMESG, typestr: string = 'mesg') {
    if (!(this.level & type)) return // bitmask

    if (typeof document !== 'undefined') {
      const logArea = document.getElementById('status')
      if (!logArea) return

      const addendum = document.createElement('span')
      addendum.setAttribute('class', typestr)
      const now = new Date()
      addendum.innerHTML = `${now.getHours()}:${now.getMinutes()}:${now.getSeconds()} [${typestr}] ${escapeForXML(str)}<br/>`
      if (!this.ascending) {
        logArea.appendChild(addendum)
      } else {
        logArea.insertBefore(addendum, logArea.firstChild)
      }
    } else if (typeof console !== 'undefined') {
      console.log(str)
    }
  }

  warn (msg: string): void {
    this.msg(msg, TWARN, 'warn')
  }

  debug (msg: string): void {
    this.msg(msg, TDEBUG, 'dbug')
  }

  info (msg: string): void {
    this.msg(msg, TINFO, 'info')
  }

  error (msg: string): void {
    this.msg(msg, TERROR, 'eror')
  }

  success (msg: string): void {
    this.msg(msg, TSUCCESS, 'good')
  }

  alert (msg: string): void {
    if (typeof alert !== 'undefined') {
      alert(msg)
    } else {
      this.warn(msg)
    }
  }

  clear (): void {
    const logArea = document.getElementById('status')
    if (!logArea) return
    logArea.innerHTML = ''
  }

  setLevel (level: number): void {
    this.level = TALL
    this.debug('Log level is now ' + level)
    this.level = level
  }

  dumpHTML (): void {
    const level = this.level
    this.level = TALL
    this.debug(document.body.innerHTML)
    this.level = level
  }
}

export const log = new Log()
