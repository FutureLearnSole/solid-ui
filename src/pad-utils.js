
const $rdf = require('rdflib')
const ns = require('./ns')
const utils = require('./utils')

const PAD = $rdf.Namespace('http://www.w3.org/ns/pim/pad#')

/* Get the chunbks of the noepad
*
* They are stored in a RDF linked list
*/
export function getChunks (subject, kb) {
  var chunks = []
  for (
    let chunk = kb.the(subject, PAD('next'));
    !chunk.sameTerm(subject);
    chunk = kb.the(chunk, PAD('next'))
  ) {
    chunks.push(chunk)
  }
  return chunks
}

/** Export a HTML string version of a notepad
* @@ no style at the moment
*
*/
export function notePadToHTML (pad, kb) {
  const chunks = getChunks(pad, kb)
  var html = '<html>\n<head>\n'
  const title = kb.anyValue(pad, ns.dct('title')) || kb.anyValue(pad, ns.dc('title'))
  if (title) {
    html += `  <title>${utils.escapeForXML(title)}</title>\n`
  }
  html += '</head>\n<body>\n'
  var level = 0
  function increaseLevel (indent) {
    for (; level < indent; level++) {
      html += '<ul>\n'
    }
  }
  function decreaseLevel (indent) {
    for (; level > indent; level--) {
      html += '</ul>\n'
    }
  }
  chunks.forEach(chunk => {
    const indent = kb.anyJS(chunk, PAD('indent'))
    const rawContent = kb.anyJS(chunk, ns.sioc('content'))
    if (!rawContent) return // seed chunk is dummy
    const content = utils.escapeForXML(rawContent)
    if (indent < 0) { // negative indent levels represenmt heading levels
      decreaseLevel(0)
      const h = indent >= -3 ? 4 + indent : 1 // -1 -> h4, -2 -> h3
      html += `\n<h${h}>${content}</h${h}>\n`
    } else { // >= 0
      if (indent > 0) { // Lists
        decreaseLevel(indent)
        increaseLevel(indent)
        html += `<li>${content}</li>\n`
      } else { // indent 0
        decreaseLevel(indent)
        html += `<p>${content}</p>\n`
      }
    }
  }) // foreach chunk
  // At the end decreaseLevel any open ULs
  decreaseLevel(0)
  html += '</body>\n</html>\n'
  return html
}
