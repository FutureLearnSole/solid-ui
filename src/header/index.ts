import { IndexedFormula, NamedNode, sym } from 'rdflib'
import { loginStatusBox, solidAuthClient } from '../authn/authn'
import { widgets } from '../widgets'
import { icon } from './icon'
import { log } from '../debug'
import { getClasses } from '../jss'
import { styleMap } from './styleMap'
// import { SolidSession } from '../../typings/solid-auth-client'
// import { emptyProfile } from './empty-profile'
// import { throttle } from '../helpers/throttle'
// import { getPod } from './metadata'
// Just for now..  ./metadata in mashlib
// 3 variables below including at_hash in Solid claim had to make camelcase...
interface SolidAuthorization {
  accessToken: string;
  clientId: string;
  idToken: string;
}

type Menu = {
  label: string,
  url: string
}
type HeaderOptions = {
  logo?: string,
  menuList?: Menu[]
}
interface SolidClaim {
  atHash: string;
  aud: string;
  azp: string;
  cnf: {
    jwk: string;
  };
  exp: number;
  iat: number;
  iss: string;
  jti: string;
  nonce: string;
  sub: string;
}
export interface SolidSession {
  authorization: SolidAuthorization;
  credentialType: string;
  idClaims: SolidClaim;
  idp: string;
  issuer: string;
  sessionKey: string;
  webId: string;
}

export function getPod (): NamedNode {
  // @@ TODO: This is given that mashlib runs on NSS - might need to change when we want it to run on other Pod servers
  return sym(document.location.origin).site()
}

export const emptyProfile = `
<svg xmlns="http://www.w3.org/2000/svg" width="26" height="26" viewBox="0 0 26 26" fill="none">
    <path fill-rule="evenodd" clip-rule="evenodd" d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z" fill="#D8D8D8" stroke="#8B8B8B"/>
    <mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="0" y="0" width="26" height="26">
        <path fill-rule="evenodd" clip-rule="evenodd" d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z" fill="white" stroke="white"/>
    </mask>
    <g mask="url(#mask0)">
        <path fill-rule="evenodd" clip-rule="evenodd" d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z" fill="#8B8B8B"/>
    </g>
</svg>`

type ThrottleOptions = {
  leading?: boolean;
  throttling?: boolean;
  trailing?: boolean;
}

export function throttle (func: Function, wait: number, options: ThrottleOptions = {}): (...args: any[]) => any {
  let context: any, args: any, result: any
  let timeout: any = null
  let previous = 0
  const later = function () {
    previous = !options.leading ? 0 : Date.now()
    timeout = null
    result = func.apply(context, args)
    if (!timeout) context = args = null
  }
  return function () {
    const now = Date.now()
    if (!previous && !options.leading) previous = now
    const remaining = wait - (now - previous)
    // @ts-ignore
    context = this
    args = arguments
    if (remaining <= 0 || remaining > wait) {
      if (timeout) {
        clearTimeout(timeout)
        timeout = null
      }
      previous = now
      result = func.apply(context, args)
      if (!timeout) context = args = null
    } else if (!timeout && options.trailing !== false) {
      timeout = setTimeout(later, remaining)
    }
    return result
  }
}

export async function initHeader (store: IndexedFormula, options: HeaderOptions) {
  const header = document.getElementById('PageHeader')
  if (!header) {
    return
  }
  const pod = getPod()
  log(pod)
  solidAuthClient.trackSession(rebuildHeader(header, store, pod, options))
}

function rebuildHeader (header: HTMLElement, store: IndexedFormula, pod: NamedNode, options: HeaderOptions) {
  return async (session: SolidSession | null) => {
    // const user = session ? sym(session.webId) : null
    // SAM - don't forget to put in the line above instead of the line below :)
    const user = sym('https://sharonstrats.inrupt.net/profile/card#me')
    log(user)
    header.innerHTML = ''
    header.appendChild(await createBanner(store, pod, user, options))
  }
}

async function createBanner (store: IndexedFormula, pod: NamedNode, user: NamedNode | null, options: HeaderOptions): Promise<HTMLElement> {
  const podLink = document.createElement('a')
  podLink.href = pod.uri
  podLink.classList.add('header-banner__link')
  podLink.innerHTML = icon

  const menu = user
    ? await createUserMenu(store, user, options)
    : createLoginSignUpButtons()

  const banner = document.createElement('div')
  banner.classList.add('header-banner')
  banner.appendChild(podLink)
  banner.appendChild(menu)

  return banner
}

function createLoginSignUpButtons () {
  const profileLoginButtonPre = document.createElement('div')
  profileLoginButtonPre.classList.add('header-banner__login')
  profileLoginButtonPre.appendChild(loginStatusBox(document, null, {}))
  return profileLoginButtonPre
}
async function openDashboardPane (outliner: any, pane: string): Promise<void> {
  outliner.showDashboard({
    pane
  })
}

function createUserMenuButton (label: string, onClick: EventListenerOrEventListenerObject): HTMLElement {
  const button = document.createElement('button')
  button.classList.add('header-user-menu__button')
  button.addEventListener('click', onClick)
  button.innerText = label
  return button
}

function getStyle (styleClass) {
  return styleMap[styleClass]
}

function addStyleClassToElement (element: any, styleClass: string) {
  const style = getStyle(styleClass)
  const { classes } = getClasses(document.head, { [styleClass]: style })
  element.classList.add(classes[styleClass])
}

function createUserMenuLink (label: string, href: string): HTMLElement {
  // if you have a uri then call that otherwise call on the button..
  // specify action in mashlib..  when you have onclick this function should be called
  // call property onclick if present it's a callback function, call upon that property
  // will outside of this function..
  const link = document.createElement('a')
  // link.classList.add('header-user-menu__link')
  const style = getStyle('header-user-menu__link')
  const { classes } = getClasses(document.head, { 'header-user-menu__link': style })
  link.classList.add(classes['header-user-menu__link'])
  link.href = href
  link.innerText = label
  return link
}

async function createUserMenu (store: IndexedFormula, user: NamedNode, options: HeaderOptions): Promise<HTMLElement> {
  const fetcher = (<any>store).fetcher
  if (fetcher) {
    // Making sure that Profile is loaded before building menu
    await fetcher.load(user)
  }
  // const outliner = getOutliner(document)
  // SAM Options: here is where I can take the list of options and add to loggedInMenuList...
  const loggedInMenuList = document.createElement('ul')
  // loggedInMenuList.classList.add('header-user-menu__list')
  addStyleClassToElement(loggedInMenuList, 'headerUserMenuList')
  loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink('Show your profile', user.uri)))
  if (options.menuList) {
    options.menuList.map(function (menuItem) {
      loggedInMenuList.appendChild(createUserMenuItem(createUserMenuLink(menuItem.label, menuItem.url)))
    })
  }
  /*
  const menuItems = await getMenuItems(outliner)
  menuItems.forEach(item => {
    loggedInMenuList.appendChild(createUserMenuItem(createUserMenuButton(item.label, () => openDashboardPane(outliner, item.tabName || item.paneName))))
  }) */
  loggedInMenuList.appendChild(createUserMenuItem(createUserMenuButton('Log out', () => solidAuthClient.logout())))

  const loggedInMenu = document.createElement('nav')
  loggedInMenu.classList.add('header-user-menu__navigation-menu')
  addStyleClassToElement(loggedInMenu, 'headerUserMenuNavigationMenu')
  loggedInMenu.setAttribute('aria-hidden', 'true')
  loggedInMenu.appendChild(loggedInMenuList)

  const loggedInMenuTrigger = document.createElement('button')
  loggedInMenuTrigger.classList.add('header-user-menu__trigger')
  loggedInMenuTrigger.type = 'button'
  /* const profileImg = getProfileImg(store, user)
  if (typeof profileImg === 'string') {
    loggedInMenuTrigger.innerHTML = profileImg
  } else {
    loggedInMenuTrigger.appendChild(profileImg)
  } */

  const loggedInMenuContainer = document.createElement('div')
  // loggedInMenuContainer.classList.add('header-banner__user-menu', 'header-user-menu')
  loggedInMenuContainer.appendChild(loggedInMenuTrigger)
  loggedInMenuContainer.appendChild(loggedInMenu)

  const throttledMenuToggle = throttle((event: Event) => toggleMenu(event, loggedInMenuTrigger, loggedInMenu), 50)
  loggedInMenuTrigger.addEventListener('click', throttledMenuToggle)
  let timer = setTimeout(() => null, 0)
  loggedInMenuContainer.addEventListener('mouseover', event => {
    clearTimeout(timer)
    throttledMenuToggle(event)
  })
  loggedInMenuContainer.addEventListener('mouseout', event => {
    timer = setTimeout(() => throttledMenuToggle(event), 200)
  })

  return loggedInMenuContainer
}

function createUserMenuItem (child: HTMLElement): HTMLElement {
  const menuProfileItem = document.createElement('li')
  menuProfileItem.classList.add('header-user-menu__list-item')
  menuProfileItem.appendChild(child)
  return menuProfileItem
}

async function getMenuItems (outliner: any): Promise<Array<{
  paneName: string;
  tabName?: string;
  label: string;
  icon: string;
}>> {
  return outliner.getDashboardItems()
}

function getProfileImg (store: IndexedFormula, user: NamedNode): string | HTMLElement {
  const profileUrl = widgets.findImage(user)
  if (!profileUrl) {
    return emptyProfile
  }
  const profileImage = document.createElement('div')
  profileImage.classList.add('header-user-menu__photo')
  profileImage.style.backgroundImage = `url("${profileUrl}")`
  return profileImage
}

function toggleMenu (event: Event, trigger: HTMLButtonElement, menu: HTMLElement): void {
  const isExpanded = trigger.getAttribute('aria-expanded') === 'true'
  const expand = event.type === 'mouseover'
  const close = event.type === 'mouseout'
  if ((isExpanded && expand) || (!isExpanded && close)) {
    return
  }
  trigger.setAttribute('aria-expanded', (!isExpanded).toString())
  menu.setAttribute('aria-hidden', isExpanded.toString())
}
