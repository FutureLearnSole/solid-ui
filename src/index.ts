/*
The MIT License (MIT)

Copyright (c) 2015-2016 Solid

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

If you would like to know more about the solid Solid project, please see
https://github.com/solid/solid
*/
'use strict'
/**
 * Provides a Solid client helper object (which exposes various static modules).
 * @module solidUi.js
 * @main solidUi.js
 */

/**
 * @class SolidUi
 * @static
 */

// REMOVE @ts-ignore as you migrate files to TypeScript
import * as rdf from 'rdflib' // pull in first avoid cross-refs
// @ts-ignore
import * as ns from './ns'
// @ts-ignore
import * as acl from './acl'
// @ts-ignore
import * as aclControl from './acl-control'
// @ts-ignore
import * as authn from './signin'
// @ts-ignore
import * as create from './create'
// @ts-ignore
import * as icons from './iconBase'
// @ts-ignore
import * as log from './log'
// @ts-ignore
import * as matrix from './matrix'
// @ts-ignore
import * as media from './media-capture'
// @ts-ignore
import * as messageArea from './messageArea'
// @ts-ignore
import { infiniteMessageArea } from './chat/infinite'
// @ts-ignore
import * as pad from './pad'
// @ts-ignore
import * as preferences from './preferences'
// @ts-ignore
import * as store from './store'
// @ts-ignore
import * as style from './style'
// @ts-ignore
import * as table from './table'
// @ts-ignore
import * as tabs from './tabs'
// @ts-ignore
import * as utils from './utils'
// @ts-ignore
import * as widgets from './widgets'

const UI = {
  ns,
  rdf,
  acl,
  aclControl,
  authn,
  create,
  dom: window.document, // Idea that UI.dom can be adapted in non-browser environments
  icons,
  log,
  matrix,
  media,
  messageArea,
  infiniteMessageArea,
  pad,
  preferences,
  store,
  style,
  table,
  tabs,
  utils,
  widgets
}

if (typeof window !== 'undefined') {
  ;(<any>window).UI = UI // Simpler access by non-node scripts
}
export default UI
