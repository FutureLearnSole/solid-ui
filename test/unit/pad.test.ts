// SHARON IS WORKING ON THIS FILE
// NOT DOING ANYTHING FOR NOW AS SHE WILL CLEAN IT UP LATER

import { JSDOM } from 'jsdom'
import * as RdfLib from 'rdflib'
import * as pad from '../../src/pad'
import widgets from '../../src/widgets'

// jest.mock('rdflib')
// jest.mock('solid-auth-client')
const window = new JSDOM('<!DOCTYPE html><p>Hello world</p>').window
const dom = window.document

describe('lightColorHash', () => {
  it('exists', () => {
    expect((pad as any).lightColorHash).toBeInstanceOf(Function)
  })
  it('returns #ffffff when an author is not provided', () => {
    // #ffffff is specifically stated in the code to be returned
    // when there is no author
    expect((pad as any).lightColorHash(null)).toBe('#ffffff')
  })
  it('returns a value when given an author', () => {
    // #dac2dc is the hash of 'https://sharonstrats.inrupt.net/profile/card#me'
    const author = { uri: 'https://sharonstrats.inrupt.net/profile/card#me' }
    expect((pad as any).lightColorHash(author)).toBe('#dac2dc')
  })
})
describe('renderPartipants', () => {
  it('exists', () => {
    expect((pad as any).renderPartipants).toBeInstanceOf(Function)
  })
  it('runs', () => {
    const table = dom.createElement('table')
    const padDoc = null
    // const subject = new RdfLib.NamedNode('test')
    const subject = null
    const me = 'webId'
    const options = {}
    expect(
      (pad as any).renderPartipants(dom, table, padDoc, subject, me, options)
    ).toMatchSnapshot()
  })
  it('returns without crashing when a person is not returned', () => {
    // @@ TODO need to mock kb.any and kb.each - not working for some reason
    // kb is a store
    const table = dom.createElement('table')
    const padDoc = null
    // const subject = new RdfLib.NamedNode('participation')
    const subject = null
    const me = 'webId'
    const options = {}
    expect(
      (pad as any).renderPartipants(dom, table, padDoc, subject, me, options)
    ).toMatchSnapshot()
  })
})
describe('participationObject', () => {
  it('exists', () => {
    expect((pad as any).participationObject).toBeInstanceOf(Function)
  })

  it('runs', () => {
    // TODO: check on arguments

    const subject = null
    const padDoc = null
    const me = null
    expect((pad as any).participationObject(subject, padDoc, me)).resolves.toBe(
      {}
    )
  })
  it('runs 2', () => {
    // TODO: check on arguments
    const spy = jest.spyOn(widgets, 'newThing')
    const subject = null
    const padDoc = document
    const me = 'https://sharonstrats.inrupt.net/profile/card#me'
    expect((pad as any).participationObject(subject, padDoc, me)).resolves.toBe(
      {}
    )
    expect(spy).toBeCalled()
  })
})
describe('recordParticipation', () => {
  it('exists', () => {
    expect((pad as any).recordParticipation).toBeInstanceOf(Function)
  })
  const subject = null
  const padDoc = null
  const refreshable = true
  it('runs', () => {
    expect((pad as any).recordParticipation(subject, padDoc, refreshable)).toBe(
      undefined
    )
  })
})
describe('manageParticipation', () => {
  it('exists', () => {
    expect((pad as any).manageParticipation).toBeInstanceOf(Function)
  })
  const container = dom.createElement('div')
  const padDoc = null
  const subject = null
  const me = null
  const options = {}
  it('runs', () => {
    expect(
      (pad as any).manageParticipation(
        dom,
        container,
        padDoc,
        subject,
        me,
        options
      )
    ).toMatchSnapshot()
  })
})
describe('notepad', () => {
  it('exists', () => {
    expect((pad as any).notepad).toBeInstanceOf(Function)
  })

  it('runs', () => {
    const padDoc = null
    const subject = null
    const me = new RdfLib.NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = {}
    expect((pad as any).notepad(dom, padDoc, subject, me, options))
      .toMatchSnapshot()
  })

  it.skip('should log error that you need to be logged in for pad to be edited', () => {
    const padDoc = null
    const subject = null
    const me = null
    const options = {}
      ; (window as any).console = { log: jest.fn() }
    expect((pad as any).notepad(dom, padDoc, subject, me, options))
      .toMatchSnapshot()
    expect(console.log).toBeCalledWith(
      'Warning: must be logged in for pad to be edited'
    )
  })

  it.skip('status area ...', () => {
    const padDoc = null
    const subject = null
    const me = { uri: 'https://sharonstrats.inrupt.net/profile/card#me' }
    const options = { statusArea: document.createElement('p') }

    expect(
      (pad as any).notepad(dom, padDoc, subject, me, options)
    ).resolves.toMatchSnapshot()
  })

  // @@ TODO the code itself seems to error where it says 'new'
  // need to research further

  it.skip('should throw an error when me is provided but no uri', () => {
    const padDoc = null
    const subject = null
    const me = {}
    const options = {}

    expect(
      (pad as any).notepad(dom, padDoc, subject, me, options)
    ).resolves.toBe({})
  })
})
