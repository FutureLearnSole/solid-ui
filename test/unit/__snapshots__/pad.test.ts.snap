// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manageParticipation runs 1`] = `
<table
  style="margin: 0.8em;"
/>
`;

exports[`notepad runs 1`] = `
<table
  style="padding: 1em; overflow: auto; resize: horizontal; min-width: 40em;"
/>
`;

exports[`renderPartipants returns without crashing when a person is not returned 1`] = `
<table
  style="margin: 0.8em;"
/>
`;

exports[`renderPartipants runs 1`] = `
<table
  style="margin: 0.8em;"
/>
`;
